### PR TITLE
FIX: Coerce `MGHImage`s to `Nifti1Image`s when plotting

### PR DESCRIPTION
**Pull request by mgxd**
Supercedes https://github.com/nipreps/smriprep/pull/317

### CODE REVIEW

**Comment by mgxd** [comment]
This fixed the errors previously encountered in the nibabies runs

**Comment by effigies** [comment]
We should report this upstream to nilearn, as well. Workaround is fine because all we care about is data/affine, but they at least used to be able to handle `SpatialImage`s in general, so this seems like a regression.

**Comment by mgxd** [comment]
@effigies - Looks like it needs to be a NiftiImage, so might as well coerce.

https://nilearn.github.io/stable/modules/generated/nilearn.plotting.plot_anat.html#nilearn.plotting.plot_anat
https://nilearn.github.io/stable/manipulating_images/input_output.html#inputing-data-file-names-or-image-objects

**Comment by effigies** [comment]
https://nilearn.github.io/stable/manipulating_images/input_output.html#niimg-like-objects

> A string or pathlib.Path object with a file path to a Nifti or Analyse image
> An SpatialImage from nibabel, ie an object exposing get_fdata() method and affine attribute, typically a Nifti1Image from [nibabel](http://nipy.sourceforge.net/nibabel/).

File: niworkflows/viz/utils.py [hunk 1]
```diff
@@ -341,11 +341,16 @@ def plot_registration(
...
 
+    anat_nii = _ensure_nii(anat_nii)
+
     out_files = []
     if estimate_brightness:
         plot_params = robust_set_limits(anat_nii.get_fdata().reshape(-1), plot_params)
 
     # FreeSurfer ribbon.mgz
+    if contour:
+        contour = _ensure_nii(contour)
```
Comment (effigies) [inline niworkflows/viz/utils.py:352]: This is just `nb.Nifti1Image.from_image()`. Let's also add a comment so we know why we did this.

```suggestion
    # nilearn 0.10.0 uses Nifti-specific methods
    anat_nii = nb.Nifti1Image.from_image(anat_nii)

    out_files = []
    if estimate_brightness:
        plot_params = robust_set_limits(anat_nii.get_fdata().reshape(-1), plot_params)

    # FreeSurfer ribbon.mgz
    if contour:
        contour = nb.Nifti1Image.from_image(anat_nii)
```

**Comment by effigies** [comment]
I'm +1 for merging rather than relying on people not to have this specific version of nilearn.

File: niworkflows/viz/utils.py [hunk 2]
```diff
@@ -714,3 +720,9 @@ def plot_melodic_components(
         pad_inches=0.01,
     )
     fig.clf()
+
+
+def _ensure_nii(img: nb.spatialimages.SpatialImage) -> nb.Nifti1Image:
+    if not isinstance(img, nb.Nifti1Image):
+        img = nb.Nifti1Image(img.dataobj, img.affine, img.header)
+    return img
```
Comment (mgxd) [inline niworkflows/viz/utils.py:728]:
```suggestion
```

**Review by effigies — APPROVED** [review]

**Pull request merged**